keps/sig-scheduling/3280-guarantee-pdb-when-preemption-happens/README.md

```diff
@@ -0,0 +1,970 @@
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+In order to address the issue mentioned above, a new field `DisruptionPolicy` will 
+be added to `PriorityClass`. Users will be able to set this field to 
+determine whether or not PDB violation will be allowed during scheduler preemption. The 
+default value for this field will be `PodDisruptionBudgetBestEffort`, which is 
+consistent with current behavior.  Additionally, in order to avoid system stability 
+issues, even if the victim's PDB is not allowed to be violated during the preemption 
+process, the preempter with priority ClassName of `system-cluster-critical` or `system-node-critical` 
+will still be able to violate the victim's PDB during preemption.
+
+```go
+type PriorityClass struct {
+	metav1.TypeMeta
+	metav1.ObjectMeta
+	Value int32
+	GlobalDefault bool
+	Description string
+	PreemptionPolicy *core.PreemptionPolicy
+
+	// DisruptionPolicy specifies how pods with this priority class 
+	// may be disrupted when being preempted by a higher priority pod.
+	// Defaults to PodDisruptionBudgetBestEffort if unset.
+	// +optional
+	DisruptionPolicy *core.DisruptionPolicy
+}
+```
+
+Currently, two simple policies are provided in this version. One is `PodDisruptionBudgetBestEffort` 
+which indicates that the scheduler will try to avoid violating the PDB during preemption, 
+but it cannot be guaranteed. The other is `PodDisruptionBudgetStrict` which indicates that the 
+PDB will not be violated during scheduler preemption, as noted above, if the preempter 
+has a priority ClassName of `system-cluster-critical` or `system-node-critical`, it may 
+still potentially violate the victim's PDB. New policies may be added in the future based 
+on the need, but it will not be discussed in this article at this time.
+
+```go
+// DisruptionPolicy specifies how pods with this priority class 
+// may be disrupted when being preempted.
+type DisruptionPolicy string
+
+const (
+	// PodDisruptionBudgetStrict indicates when being preempted by a higher priority pod, 
+	// strict adherence to the PodDisruptionBudget of the pod with this priority class is required.
+	// But it the preempter with these priority classes of `system-cluster-critical` or `system-node-critical` 
+	// will still be able to violate this pod's PDB during preemption.
+	PodDisruptionBudgetStrict PreemptedPolicy = "PodDisruptionBudgetStrict"
```

Comment:
`SystemCriticalOnly`

Reply:
to consider: https://github.com/kubernetes/enhancements/pull/3755/files#r1091298934

Reply:
nit: `PDBStrict DisruptionPolicy = ...`